Samples/SimpleInk/cpp/Scenario13.xaml.cpp

```diff
@@ -0,0 +1,410 @@
+
+        ResizeSelectedInkStrokes(scale, *center);
+        break;
+    }
+}
+
+void Scenario13::SelectionRectangle_ManipulationCompleted(Object^ sender, Windows::UI::Xaml::Input::ManipulationCompletedRoutedEventArgs^ args)
+{
+    currentManipulationType = ManipulationTypes::None;
+    Windows::UI::Core::CoreWindow^ window = Windows::UI::Core::CoreWindow::GetForCurrentThread();
+    window->PointerCursor = ref new CoreCursor(CoreCursorType::Arrow, 0);    
+}
+
+void Scenario13::SelectionRectangle_PointerPressed(Object^ sender, Windows::UI::Xaml::Input::PointerRoutedEventArgs^ args)
+{
+    if (currentManipulationType != ManipulationTypes::None)
+        return;
+
+    Windows::UI::Core::CoreWindow^ window = Windows::UI::Core::CoreWindow::GetForCurrentThread();
+
+    if (window->PointerCursor->Type == CoreCursorType::SizeAll)
+    {
+        currentManipulationType = ManipulationTypes::Move;
+    }
+    else
+    {
+        currentManipulationType = ManipulationTypes::Size;
+
+        switch (resizePointerCornor)
+        {
+        case RectangleCorner::TopLeft:
+        case RectangleCorner::BottomRight:
+            window->PointerCursor = ref new CoreCursor(CoreCursorType::SizeNorthwestSoutheast, 0);
+            break;
+        case RectangleCorner::TopRight:
+        case RectangleCorner::BottomLeft:
+            window->PointerCursor = ref new CoreCursor(CoreCursorType::SizeNortheastSouthwest, 0);
+            break;
+        }
+    }
+}
+
+void Scenario13::SelectionRectangle_PointerMoved(Object^ sender, Windows::UI::Xaml::Input::PointerRoutedEventArgs^ args)
+{
+    if (currentManipulationType != ManipulationTypes::None)
+        return;
+
+    Windows::UI::Core::CoreWindow^ window = Windows::UI::Core::CoreWindow::GetForCurrentThread();
+    Point mousePos = args->GetCurrentPoint(selectionRectangle)->Position;
+
+    SetPointerCursorType(mousePos);
+}
+
+void Scenario13::SelectionRectangle_PointerExited(Object^ sender, Windows::UI::Xaml::Input::PointerRoutedEventArgs^ args)
+{
+    if (currentManipulationType != ManipulationTypes::None)
+        return;
+
+    Windows::UI::Core::CoreWindow^ window = Windows::UI::Core::CoreWindow::GetForCurrentThread();
+    window->PointerCursor = ref new CoreCursor(CoreCursorType::Arrow, 0);
+}
+
+/// <summary>
+/// The method will take all Strokes from the inkCanvas that have been "Selected" as per the link statement.
+/// Then we will apply a Matrix3x2 transform, based on the supplied scale and the topLeft point of the bounding box.
+/// </summary>
+/// <param name="scale">Scaling factor to apply to the selected points</param>
+/// <param name="topLeft">the "X,Y" coordinates of the Selection Box. </param>
+void Scenario13::ResizeSelectedInkStrokes(float scale, Windows::Foundation::Numerics::float2& topLeft)
+{
+
+    auto allStrokes = inkCanvas->InkPresenter->StrokeContainer->GetStrokes();
+
+    if (allStrokes == nullptr)
```

Comment:
winrt containers should never be null. instead empty (Size() == 0), is this an API bug?